### PR TITLE
Q-plane interpolation and Q-plane spectrogram plotting mechanisms

### PR DESCRIPTION
**Pull request by hagabbar**
1.) Addition of loudest Q-plane interpolation function (defined as Q-plane with maximum normalized Q-tile energy)
2.) Addition of Q-plane spectrogram plotting function.

### CODE REVIEW

File: pycbc/filter/qtransform.py [hunk 1]
```diff
@@ -32,12 +32,156 @@
...
 from numpy import pi, ceil, log, exp
 import numpy as np
 from pycbc.strain  import next_power_of_2
 from pycbc.types.timeseries import FrequencySeries, TimeSeries
+from scipy.interpolate import (interp2d, InterpolatedUnivariateSpline)
+from pycbc.fft import ifft
+from pycbc.types import zeros
 from numpy import fft as npfft
 
+from matplotlib import use
```
Comment (ahnitz) [inline pycbc/filter/qtransform.py]: It's not a great idea to import matplotlib from within the library. If you need it. Please import from the function you'll use it in.

File: pycbc/filter/qtransform.py [hunk 2]
```diff
@@ -32,12 +32,156 @@
...
+       tile with the maximum normalized energy. Q-transform is then 
+       interpolated to a desired frequency and time resolution.
+
+    Parameters
+    ----------
+    qplane_tile_dict:
+        Dictionary containing a list of q-tile tupples for each q-plane
+    fseries: 'pycbc FrequencySeries'
+        frequency-series data set
+    sampling:
```
Comment (ahnitz) [inline pycbc/filter/qtransform.py]: I don't think you need sampling as an argument. That can be derived from the fseries itself. sample_rate = (len(fseries) -1) * 2 * df

File: pycbc/filter/qtransform.py [hunk 3]
```diff
@@ -32,12 +32,156 @@
...
+                        z, cmap=cmap, norm=None)
+    plt.colorbar(p1)
+    plt.title('Pycbc q-transform')
+
+    plt.savefig('%s/run_%s/spec.png' % (out_dir,now))
+
+    return plt
+
+def Qplane(qplane_tile_dict, fseries, sampling, out_dir, now, frange, normalized=True, tres=0.001, fres=1.):
+    """Performs q-transform on each tile for each q-plane and selects
```
Comment (ahnitz) [inline pycbc/filter/qtransform.py]: Lower case is the standard for functions.

File: pycbc/filter/qtransform.py [hunk 4]
```diff
@@ -32,12 +32,156 @@
...
+    ----------
+    qplane_tile_dict:
+        Dictionary containing a list of q-tile tupples for each q-plane
+    fseries: 'pycbc FrequencySeries'
+        frequency-series data set
+    sampling:
+        sampling rate of data set
+    normalized: 'bool'
+        normalize energy time series?
+    out_dir:
```
Comment (ahnitz) [inline pycbc/filter/qtransform.py]: out_dir is not used in this function

Comment (ahnitz) [inline pycbc/filter/qtransform.py]: Nor is now.

File: pycbc/filter/qtransform.py (hunk 2 above)
Comment (hagabbar) [inline pycbc/filter/qtransform.py]: How is df defined?

Comment (hagabbar) [inline pycbc/filter/qtransform.py]: Or does the pycbc frequency series have an attribute .sample_rate?

Comment (hagabbar) [inline pycbc/filter/qtransform.py]: Ah wait, df is just fseries.delta_f

Comment (hagabbar) [inline pycbc/filter/qtransform.py]: @ahnitz What if the user wants to use a different sampling rate than that of the channel? Can I just set the variable 'sampling' to some default value in the definition of the function?

Comment (hagabbar) [inline pycbc/filter/qtransform.py]: @ahnitz Or should we just assume that the user is going to want to sample at that channel's native frequency?

Comment (hagabbar) [inline pycbc/filter/qtransform.py]: I think we should leave it in as an option, but if the user doesn't explicitly specify then we just set it to the default native sampling rate.

Comment (hagabbar) [inline pycbc/filter/qtransform.py]: On second thought, the user can just write a wrapper to downsample. I'll just remove those options as you requested earlier.

**Pull request merged**